src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java

```diff
@@ -135,6 +137,15 @@
     + "Defaults to the same dataset in which the view is located.")
   private String viewMaterializationDataset;
 
+  @Name(TEMP_TABLE_EXPIR_UNIT)
+  @Nullable
+  @Description("Amount of time before expiration of a temp table created to filter BigQuery")
+  private String tempTableExpirationTimeUnit;
+
+  @Name(TEMP_TABLE_EXPIR_NUM)
+  @Nullable
+  @Description("Amount of time before expiration of a temp table created to filter BigQuery")
+  private String tempTableExpirationTimeNumber;
```

Comment:
why is this a string?

Reply:
Default value for the get method on configuration requires a String

Reply:
this can be `private Boolean` no?

Reply:
this number no since its going to be between 1 - 30. But the Unit above I can make a boolean since its 2 values, Day and Hour. I'll switch the Unit to Boolean for reduced memory usage.

Reply:
this config no since its going to be between 1 - 30. But the Unit above could be a boolean since its 2 values, Day and Hour. I could redo the Unit and use setBoolean for the config to save the memory allocation of a String.

Reply:
No. Looks good.

Reply:
Looks good.

Reply:
Actually this can be an integer.